packages/react-aria-components/src/Menu.tsx

```diff
@@ -179,6 +179,16 @@ function MenuInner<T extends object>({props, collection, menuRef: ref}: MenuInne
     isDisabled: isSubmenu || rootMenuTriggerState?.UNSTABLE_expandedKeysStack.length === 0
   });
 
+  let prevPopoverContainer = useRef<HTMLDivElement | null>(null) ;
+  let [leftOffset, setLeftOffset] = useState({left: 0});
+  useEffect(() => {
+    if (popoverContainer && prevPopoverContainer.current !== popoverContainer && leftOffset.left === 0) {
+      prevPopoverContainer.current = popoverContainer;
+      let {left} = popoverContainer.getBoundingClientRect();
+      setLeftOffset({left: -1 * left});
+    }
+  }, [leftOffset, popoverContainer]);
+
```

Comment:
The div that serves as the popoverContainer has `width: 100vw` (aka full page width) but its origin is set as the left edge of its parent menu, resulting in a div that extends far off the page depending on the parent menu's positioning. The above copies the same logic from RSP Submenu which calculates how much said container needs to be offset in order to make it start from the left most edge of the page instead, meaning it will be flush with the boundaries of the page. This give the positioning code the proper container "area" in which it can calculate if the submenu should be repositioned/flipped/sized.